src/components/manager.ts

```diff
@@ -865,13 +866,33 @@ export class Manager {
         }
     }
 
+    private kpsewhichBibPath(bib: string): string | undefined {
+        const kpsewhich = vscode.workspace.getConfiguration('latex-workshop').get('kpsewhich.path') as string
+        try {
+            const kpsewhichReturn = cs.sync(kpsewhich, ['-format=.bib', bib])
+            if (kpsewhichReturn.status === 0) {
+                const bibPath = kpsewhichReturn.stdout.toString().replace(/\r?\n/, '')
+                if (bibPath === '') {
+                    return undefined
+                } else {
+                    this.extension.logger.addLogMessage(`Found .bib file using kpsewhich: ${bibPath}`)
+                    return bibPath
+                }
+            }
+        } catch(e) {
+            this.extension.logger.addLogMessage(`Cannot run kpsewhich to resolve .bib file: ${bib}`)
+        }
+        return undefined
+    }
+
     private resolveBibPath(bib: string, rootDir: string) {
         const bibDirs = vscode.workspace.getConfiguration('latex-workshop').get('latex.bibDirs') as string[]
         const bibPath = utils.resolveFile([rootDir, ...bibDirs], bib, '.bib')
 
         if (!bibPath) {
             this.extension.logger.addLogMessage(`Cannot find .bib file: ${bib}`)
-            return undefined
+            this.extension.logger.addLogMessage('Trying to use kpsewhich')
+            return this.kpsewhichBibPath(bib)
```

Comment:
I am not sure it is a good idea calling kpsewhich by default.

Reply:
I could add a new setting to use `kpsewhich` and completely ignore the `bibDirs` setting.

Reply:
In the end, I think it is better to always use the `bibDirs` setting and to only use `kpsewhich` when `latex-workshop.kpsewhich.enabled` is true (default value is false).